config/initializers/rack_attack.rb

```diff
@@ -256,6 +256,7 @@ def headers
   'rack.attack',
 ) do |_name, _start, _finish, _request_id, payload|
   req = payload[:request]
+  next if req.env['rack.attack.match_type'] == :safelist
   user = req.env['warden'].user || AnonymousUser.new
```

Comment:
The reason this change is necessary is because the session limitable behavior is very eager in prompting the redirect, and any reference to the `user` (such as on line 260 here) will trigger [the `after_set_user` hook](https://github.com/18F/identity-idp/blob/61c5e6685f87ee383941d4924ecabd1fab823513/config/initializers/session_limitable.rb#L16) and force a redirect. This was causing related tests to fail because the safelisting triggered this subscription. Since this subscriber callback is intended to capture rate limits being triggered, we shouldn't bother with it for the safelist scenario.

Reply:
would we still want the logging for this here? Like would it keep more desirable logign behavior if we did something like....
```ruby
user = req.env['warden'].user if req.env['rack.attack.match_type'] != :safelist
user ||= AnonymousUser.new
```

Reply:
My hunch was that we only want the logging if a rate limit is being triggered, and this safelist exception wouldn't fall within that.

Maybe it'd be clearer if I inverted the condition to check for `next if req.env['rack.attack.match_type'] != :blocklist`?

Reply:
yeah I think that would definitely be clearear

Reply:
Looking closer, there's a handful of different match types ([`blocklist`](https://github.com/rack/rack-attack/blob/1f57b3361e6db633f9906d823cde0268d9492870/lib/rack/attack/blocklist.rb#L8), [`safelist`](https://github.com/rack/rack-attack/blob/1f57b3361e6db633f9906d823cde0268d9492870/lib/rack/attack/safelist.rb#L8), [`throttle`](https://github.com/rack/rack-attack/blob/1f57b3361e6db633f9906d823cde0268d9492870/lib/rack/attack/throttle.rb#L18), [`track`](https://github.com/rack/rack-attack/blob/1f57b3361e6db633f9906d823cde0268d9492870/lib/rack/attack/track.rb#L9)), and it seems safer to me to fail open to log everything except `:safelist` vs. trying to list out all the types we care for, i.e. leave it as-is. WDYT?

Reply:
except `:safelist` seems fine